Instructions view for navigation

Kraci opened the pull request:
- direction instructions
- signpost instructions
- none instructions
- pdf vector arrow directions

Kraci committed:
Instructions view for navigation

- direction instructions
- signpost instructions
- none instructions
- pdf vector arrow directions

Kraci committed:
Instruction views landscape support

Kraci committed:
Instructions view landscape fix

Kraci committed:
Instruction texts if next road isn't available

jcaliksygic commented on Example/Podfile:
```diff
@@ -7,6 +7,7 @@ def shared_pods
     pod 'SygicUIKit', :git => 'https://github.com/Sygic/sygic-ui-kit-ios.git', :branch => 'develop', :commit => 'HEAD'
     pod 'SygicMaps'
     pod 'SygicMapsKit', :path => '../'
+    pod 'SwiftSVG', :git => 'https://github.com/anonymcek/SwiftSVG.git'
```
toto teda vieme vyhodit ?

Kraci replied:
dufam ano, idem sa na to teraz pozriet

jcaliksygic commented on Example/SygicMapsKit/Navigation examples/NavigationSignpostsExampleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
```
```suggestion
        view.backgroundColor = .background
```

jcaliksygic commented on Example/SygicMapsKit/Navigation examples/NavigationSignpostsExampleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        setupInitializingActivityIndicator()
```
toto je nakoniec asi zbytocne a treba to vyhodit aj z ostatnych examplov. ty si vies prezentnut prazdny navigation modul bez routy a ked ti pride compute s routov tak len na module zavolas startNavigation(with: route)

jcaliksygic commented on SygicMapsKit.podspec:
```diff
@@ -24,9 +24,11 @@ Pod::Spec.new do |s|
 
   s.dependency 'SygicMaps'#, '~> 13.0'
   s.dependency 'SygicUIKit', '~> 2.0'
+  s.dependency 'SwiftSVG'
```
hm? :)

jcaliksygic commented on SygicMapsKit/Classes/Extensions/SYManeuverExtensions.swift:
```diff
@@ -0,0 +1,199 @@
+/// Maneuver extensions for mapping arrow directions to vector images from assets catalog.
+public extension SYManeuver {
+    
+    @objc func toImage() -> String {
```
toImage by som premenoval na imageName alebo zmenil return value zo Stringu na Image rovno

jcaliksygic commented on SygicMapsKit/Classes/NavigationModule/SYMKNavigationView.swift:
```diff
@@ -87,10 +100,27 @@ public class SYMKNavigationView: UIView {
+    public func setupInstructionView(_ instructionView: SYMKInstructionView?) {
+        self.instructionView?.removeFromSuperview()
+        guard let instructionView = instructionView else { return }
+        self.instructionView = instructionView
```
toto hod este nad guard ak to je optional aby si to schoval ked dojde nil. takto ti tam ostane zbytocne referencia na view

jcaliksygic commented on SygicMapsKit/Classes/NavigationModule/SYMKNavigationView.swift:
```diff
@@ -87,10 +100,27 @@ public class SYMKNavigationView: UIView {
+        instructionView.leadingAnchor.constraint(equalTo: safeLeadingAnchor, constant: margin).isActive = true
+        landscapeInstructionWidthAnchor = instructionView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: 0.4)
+        trailingInstructionAnchor = instructionView.trailingAnchor.constraint(equalTo: safeTrailingAnchor, constant: -margin)
+        trailingInstructionAnchor?.isActive = true
```
tu by som bol opatrny a skontroloval ci som v portrete alebo landscape a podla toho aktivoval danu constraintu

jcaliksygic commented on SygicMapsKit/Classes/NavigationModule/SYMKNavigationView.swift:
```diff
@@ -59,6 +64,14 @@ public class SYMKNavigationView: UIView {
+        
+        let isLandscape = SYUIDeviceOrientationUtils.isLandscape()
+        trailingInstructionAnchor?.isActive = !isLandscape
+        landscapeInstructionWidthAnchor?.isActive = isLandscape
```
zrobil by som metodku:
func updateInstructionViewConstriants(for landscape: Bool)

Kraci committed:
Update Example/SygicMapsKit/Navigation examples/NavigationSignpostsExampleViewController.swift

Co-Authored-By: jcaliksygic <30600531+jcaliksygic@users.noreply.github.com>

Kraci committed:
Follow distance threshold for direction view

Kraci committed:
Merge branch 'MS-6022-signposty-v-navigacnom-fragmente' of https://github.com/Sygic/sygic-maps-kit-ios into MS-6022-signposty-v-navigacnom-fragmente

* 'MS-6022-signposty-v-navigacnom-fragmente' of https://github.com/Sygic/sygic-maps-kit-ios:
  Update Example/SygicMapsKit/Navigation examples/NavigationSignpostsExampleViewController.swift

Kraci committed:
Pdf signpost symbols

- swiftSVG remove
- vector pdf support for road signs and pictograms
- dont redraw signpost symbols if they are the same

Kraci committed:
iPad action sheet popover fix

Kraci committed:
Remove loading indicator in navigation examples

- instruction type button in navigation example changed to BarButtonItem

jcaliksygic committed:
example error message alert

Kraci committed:
Failing test comment
